### PR TITLE
sig-security: 2017-05-24 meeting notes

### PR DESCRIPTION
**Pull request by tych0**
Here are the meeting notes from the recent SIG-security meeting.

### CODE REVIEW

File: reports/sig-security/2017-05-24.md [hunk 1]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
   - we can propose additional deep dives and discussion topics!
 
 ## Meeting Notes
+
+* Administrivia
+  * There is a code of conduct
+  * Attendees from Docker, Intel, HP, Google, IBM, ARM, Arksan (sp?) technologies
```
Comment (riyazdf) [inline reports/sig-security/2017-05-24.md]: nits: HPE, Arxan

File: reports/sig-security/2017-05-24.md [hunk 2]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+    building tool, not a distro itself
+  * Grew out of Docker for \* ({AWS, Mac, etc.})
+  * Borrowed userspace mostly from Alpine
+  * system daemons (e.g. DHCP, possibly SSH, etc.) run in containers, which are
+    distributed as Docker images
+  * base OS is immutable, since daemons are containers
+* Projects
+  * Clear Containers
+    * Question: what's the Intel feeling r.e. kvmtool, are they still
+      interested in using it for clear containers?
```
Comment (riyazdf) [inline reports/sig-security/2017-05-24.md]: cc @jcvenegas @mcastelino - @zx2c4 asked this question and we would like to hear your input if you can give any :)

File: reports/sig-security/2017-05-24.md [hunk 3]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+* What is LinuxKit?
+  * LinuxKit is a toolkit for building container-focused Linuxen. i.e. distro
+    building tool, not a distro itself
+  * Grew out of Docker for \* ({AWS, Mac, etc.})
+  * Borrowed userspace mostly from Alpine
+  * system daemons (e.g. DHCP, possibly SSH, etc.) run in containers, which are
+    distributed as Docker images
+  * base OS is immutable, since daemons are containers
+* Projects
+  * Clear Containers
```
Comment (riyazdf) [inline reports/sig-security/2017-05-24.md]: could we make these titles link to the project subdirectories in the repo?  Might make it easier for folks to find them

File: reports/sig-security/2017-05-24.md [hunk 4]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+    * kernel module for now, working on upstreaming
+    * exposes a network device, so everything going through it is secure
+  * IMA namespacing
+    * IMA itself is designed to detect any changes to files
+    * allows users to specify policies about which files to check
+    * EVM protects changes to file xattrs, etc.
+    * IMA is not namespace aware right now, the goal is to be able to add
+      custom policies per-mount-namespace policies
+* "hardened" channel
+  * maybe don't call it "hardened", since it really means "testing" (staging,
```
Comment (riyazdf) [inline reports/sig-security/2017-05-24.md]: could we add @zx2c4's point that "hardened" also makes it sound like the "mainstream" linuxkit is insecure?

File: reports/sig-security/2017-05-24.md [hunk 5]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+    probational)
+  * require CI for graduation
+* wrap up
+  * forum link above
+  * video recording: (TBD)
```
Comment (riyazdf) [inline reports/sig-security/2017-05-24.md]: the video is still processing.  I can open a followup PR with the link and also a pointer to the next meeting's agenda

File: reports/sig-security/2017-05-24.md [hunk 6]
```diff
@@ -26,3 +26,60 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+      SELinux can also solve
+    * no assumptions about policy, subjects, objects, etc. made by other LSMs
+  * LSM stacking
+    * hopefully this decade :)
+    * previous versions went up to a v22, but progress being made
+  * mirageSDK
+    * re-write system daemons that have lots attack surface but don't get much
+      attention (dhcpd is a great example, needs privs for netlink and such)
+    * dhcpd works (used in Docker desktop client)
+    * hoping to submit to google clusterfuzz
```
Comment (NathanMcCauley) [inline reports/sig-security/2017-05-24.md]: @avsm maybe someone from clusterfuzz could present to us at linuxkit security SIG in the future?

File: reports/sig-security/2017-05-24.md [hunk 7]
```diff
@@ -26,3 +26,61 @@ Announcement: [Moby project forum post](https://forums.mobyproject.org/t/introdu
...
+    building tool, not a distro itself
+  * Grew out of Docker for \* ({AWS, Mac, etc.})
+  * Borrowed userspace mostly from Alpine
+  * system daemons (e.g. DHCP, possibly SSH, etc.) run in containers, which are
+    distributed as Docker images
+  * base OS is immutable, since daemons are containers
+* Projects
+  * [Clear Containers](../../projects/clear-containers/)
+    * Question: what's the Intel feeling r.e. kvmtool, are they still
+      interested in using it for clear containers?
```
Comment (zx2c4) [inline reports/sig-security/2017-05-24.md]: To add some context here... `qemu` is big and bloated and has a scary attack surface. `kvmtool` promised to be a super minimal hypervisor, only emulating virtio devices, to reduce attack surface and also increase boot time considerably. It looked like Intel clear containers initially went this route, but then they switched gears to pursue some kind of qemu-integrated-orchestration monstrosity. I have no idea what became of all of this, but I'd be quite happy to learn that `kvmtool` will continue to have resources poured into keeping it lean, minimal, functional, and secure.

**Comment by riyazdf** [comment]
LGTM - I've pinged @mcastelino on the community slack as well.  I'm fine merging as-is and updating with any insight he can provide.

**Pull request merged**